main/src/library/Fixpoint/Ast/Constraint.flix

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 Benjamin Dahse
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Fixpoint/Ast {
+    pub enum Constraint[v] {
+        case Constraint(HeadPredicate[v], Array[BodyPredicate[v]], SourceLocation)
+    }
+
+    instance ToString[Constraint[v]] with ToString[v] {
+        pub def toString(c: Constraint[v]): String =
+            use Fixpoint.commaSeparate;
+            match c {
+                case Constraint(head, body, _) =>
+                    if (Array.length(body) == 0)
```

Comment:
Yeah the separation of facts and rules makes it unnecessary. In a case like this, a `isFact` function would not really be helpful since I still need to match on `c` to extract `head` and `body`.